pkg/kapis/iam/v1alpha2/register.go

```diff
@@ -41,135 +38,59 @@ const groupName = "iam.kubesphere.io"
 	ws := runtime.NewWebService(GroupVersion)
 
 	handler := newIAMHandler(k8sClient, factory, ldapClient, cacheClient, options)
 
-	ws.Route(ws.POST("/authenticate").
-		To(handler.TokenReviewHandler).
-		Doc("TokenReview attempts to authenticate a token to a known user. Note: TokenReview requests may be cached by the webhook token authenticator plugin in the kube-apiserver.").
-		Reads(iamv1alpha2.TokenReview{}).
-		Returns(http.StatusOK, api.StatusOK, iamv1alpha2.TokenReview{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.POST("/login").
-		To(handler.Login).
-		Doc("KubeSphere APIs support token-based authentication via the Authtoken request header. The POST Login API is used to retrieve the authentication token. After the authentication token is obtained, it must be inserted into the Authtoken header for all requests.").
-		Reads(iamv1alpha2.LoginRequest{}).
-		Returns(http.StatusOK, api.StatusOK, models.AuthGrantResponse{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.POST("/users").
-		To(handler.CreateUser).
-		Doc("Create a user account.").
-		Reads(iamv1alpha2.CreateUserRequest{}).
-		Returns(http.StatusOK, api.StatusOK, iamv1alpha2.UserDetail{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.DELETE("/users/{user}").
-		To(handler.DeleteUser).
-		Doc("Delete the specified user.").
-		Param(ws.PathParameter("user", "username")).
-		Returns(http.StatusOK, api.StatusOK, errors.Error{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.PUT("/users/{user}").
-		To(handler.ModifyUser).
-		Doc("Update information about the specified user.").
-		Param(ws.PathParameter("user", "username")).
-		Reads(iamv1alpha2.ModifyUserRequest{}).
-		Returns(http.StatusOK, api.StatusOK, errors.Error{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.GET("/users/{user}").
-		To(handler.DescribeUser).
-		Doc("Describe the specified user.").
-		Param(ws.PathParameter("user", "username")).
-		Returns(http.StatusOK, api.StatusOK, iamv1alpha2.UserDetail{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.GET("/users").
-		To(handler.ListUsers).
-		Doc("List all users.").
-		Returns(http.StatusOK, api.StatusOK, iamv1alpha2.ListUserResponse{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.IdentityManagementTag}))
-	ws.Route(ws.GET("/users/{user}/roles").
-		To(handler.ListUserRoles).
-		Doc("Retrieve all the roles that are assigned to the specified user.").
-		Param(ws.PathParameter("user", "username")).
-		Returns(http.StatusOK, api.StatusOK, []*rbacv1.Role{}).
-		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AccessManagementTag}))
+	// implemented by create CRD object.
+	ws.Route(ws.POST("/users"))
```

Comment:
Unfinished?

Reply:
This API has been deprecated, and then users are created through CRD.